libraries/AP_Common/AP_Common.h

```diff
@@ -77,6 +77,8 @@
   bool. Bitnumber starts at 0 for the first bit
  */
 #define BIT_IS_SET(value, bitnumber) (((value) & (1U<<(bitnumber))) != 0)
+#define BIT_SET(value, bitnumber) ((value) |= (((typeof(value))1U) << (bitnumber)))
```

Comment:
I would prefer some template version like 
template <typename T>
```
void BIT_SET (T& value, uint8_t bitnumber) noexcept {
    static_assert(std::is_integral<T>::value, "Integral required.");
    ((value) |= ((T)(1U) << (bitnumber)));
}
```

The code will be the same on non debug version and we can put some assert even if the compiler is smart enough to detect issue when float is used here ! https://gcc.godbolt.org/z/8jMcfv
typeof is gcc specific and shouldn't be used IMHO

Reply:
right now it will not fit even on the KakuteF7 since it has all the features re-enabled   3KB left.....
F7 1MB boards have 32KB less space than the F4 boards due to memory sector reserved requirements....
would have to remove a LOT of stuff on that board to fit this...

Reply:
also, even if many want a feature set, changing the hwdef would break many existing vehicles....HAL_MINIMIZE_FEATURES was done a while back....before those features prolierated....the newer 1MB boards had esoteric features removed like parachute and gripper (and still do) and we eliminated EKF2 (doesnt break an existing EKF2 use), but that is almost gone now....~ 10K in the F4s now
I think that this is going to wind up being a 2KB board feature only....I wish we had scripting on the 1MB boards, but, oh well, cant happen

Reply:
> template <typename T>
> 
> void BIT_SET (T& value, uint8_t bitnumber) noexcept {
>     static_assert(std::is_integral<T>::value, "Integral required.");
>     ((value) |= ((T)(1U) << (bitnumber)));
> }

cool! mine was a bit rougher ;)